src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.cs

```diff
@@ -4208,6 +4208,112 @@ private static string GetCommentExpectedString(bool prettyPrint)
             return Encoding.UTF8.GetString(ms.ToArray());
         }
 
+        [Fact]
+        public void WriteCommentsInArrayAsIndentedJson()
+        {
+            const string expectedIndentedJson =
+@"/*Comment at start of doc*/
+[
+  /*Comment as first array item*/
+  []
+  /*Comment in the middle of array*/,
+  {}
+  /*Comment as the last array item*/
+]
+/*Comment at end of doc*/";
+
+            var options = new JsonWriterOptions { Indented = true, SkipValidation = false };
+
+            for (int i = 0; i < 3; i++)
+            {
+                var output = new ArrayBufferWriter<byte>(32);
+                using var jsonUtf8 = new Utf8JsonWriter(output, options);
+
+                WriteCommentValue(jsonUtf8, i, "Comment at start of doc");
+                jsonUtf8.WriteStartArray();
+
+                WriteCommentValue(jsonUtf8, i, "Comment as first array item");
+
+                jsonUtf8.WriteStartArray();
+                jsonUtf8.WriteEndArray();
+
+                WriteCommentValue(jsonUtf8, i, "Comment in the middle of array");
+
+                jsonUtf8.WriteStartObject();
+                jsonUtf8.WriteEndObject();
+
+                WriteCommentValue(jsonUtf8, i, "Comment as the last array item");
+
+                jsonUtf8.WriteEndArray();
+
+                WriteCommentValue(jsonUtf8, i, "Comment at end of doc");
+
+                jsonUtf8.Flush();
+
+                JsonTestHelper.AssertContents(expectedIndentedJson, output);
+            }
+        }
+
+        [Fact]
+        public void WriteCommentsInObjectAsIndentedJson()
+        {
+            const string expectedIndentedJson =
+@"/*Comment at start of doc*/
```

Comment:
You can use the new raw string literals feature for more readable JSON:

https://devblogs.microsoft.com/dotnet/csharp-11-preview-updates/#raw-string-literals

Reply:
Changed.